File: app/src/main/java/com/example/markkko/povezime/app/util/StringUtils.java
package com.example.markkko.povezime.app.util;

import android.text.TextUtils;
import android.util.Patterns;
import android.widget.EditText;
import android.widget.TextView;

import java.util.regex.Matcher;
import java.util.regex.Pattern;

public class StringUtils {

	public static boolean isNullOrEmpty(final String string) {
		return string == null || "".equals(string);
	}

	public static boolean isNullOrEmpty(final CharSequence string) {
		return string == null || "".equals(string);
	}

	public static boolean isNullOrEmpty(final TextView input) {
		return input == null || input.getText().toString().equals("");
	}

	public static boolean isNullOrEmpty(final EditText input) {
		return input == null || input.getText().toString().equals("");
	}

	public static boolean isNumberBad(final EditText input) {
		if (input == null) {
			return true;
		}

		try {
			float number = Float.parseFloat(input.getText().toString());
			return number < 0;
		} catch (Exception e) {
			return true;
		}
	}

	public static float getFloatSafe(final EditText input) {
		try {
			return Float.parseFloat(input.getText().toString());
		} catch (Exception e) {
			return 0;
		}
	}

	public static String getStringSafe(final TextView input) {
		if (input == null || input.getText() == null) return "";
		return getStringSafe(input.getText().toString());
	}

	public static String getStringSafe(final EditText input) {
		if (input == null || input.getText() == null) return "";
		return getStringSafe(input.getText().toString());
	}

	public static String getStringSafe(final String input) {
		if (isNullOrEmpty(input)) {
			return "";
		}
		return input;
	}

	public static String getStringSafe(final String input, final String replacement) {
		if (isNullOrEmpty(input)) {
			return replacement;
		}
		return input;
	}

	public static boolean isValidEmail(EditText target) {
		return target != null && android.util.Patterns.EMAIL_ADDRESS.matcher(target.getText()).matches();
	}

	public static boolean isValidEmail(CharSequence target) {
		return target != null && android.util.Patterns.EMAIL_ADDRESS.matcher(target).matches();
	}

	public static boolean isValidPhoneNumber(CharSequence phoneNumber) {
		return !TextUtils.isEmpty(phoneNumber) && android.util.Patterns.PHONE.matcher(phoneNumber).matches();
	}

	public static boolean isValidWebsite(EditText target) {
		return !isNullOrEmpty(target) && Patterns.WEB_URL.matcher(target.getText()).matches();
	}

	public static boolean isValidWebsite(CharSequence target) {
		return !TextUtils.isEmpty(target) && Patterns.WEB_URL.matcher(target).matches();
	}

	public static boolean passwordCharValidation(String password) {
		if (password.length() < 8) {
			return false;
		}
		String PASSWORD_PATTERN = "^(?=.*[A-Z])(?=.*[@_.]).*$";
		Pattern pattern = Pattern.compile(PASSWORD_PATTERN);
		Matcher matcher = pattern.matcher(password);
		return !password.matches(".*\\d.*") || !matcher.matches();
	}

	public static boolean passwordValidationShort(String password) {
		return password.length() > 8;
	}
}

File: app/src/main/java/com/example/markkko/povezime/app/util/StringUtils.kt
package com.example.markkko.povezime.app.util

import android.text.TextUtils
import android.util.Patterns
import android.widget.EditText
import android.widget.TextView
import java.util.regex.Pattern

fun isNullOrEmpty(string: String?): Boolean {
    return string == null || "" == string
}

fun isNullOrEmpty(string: CharSequence?): Boolean {
    return string == null || "" == string
}

fun isNullOrEmpty(input: TextView?): Boolean {
    return input == null || input.text.toString() == ""
}

fun isNullOrEmpty(input: EditText?): Boolean {
    return input == null || input.text.toString() == ""
}

fun isNumberBad(input: EditText?): Boolean {
    if (input == null) {
        return true
    }

    try {
        val number = java.lang.Float.parseFloat(input.text.toString())
        return number < 0
    } catch (e: Exception) {
        return true
    }

}

fun getFloatSafe(input: EditText): Float {
    return try {
        java.lang.Float.parseFloat(input.text.toString())
    } catch (e: Exception) {
        0f
    }
}

fun getIntSafe(input: EditText): Int {
    return try {
        input.text.toString().toInt()
    } catch (e: Exception) {
        0
    }
}

fun getStringSafe(input: TextView?): String {
    return if (input == null || input.text == null) "" else getStringSafe(input.text.toString())
}

fun getStringSafe(input: EditText?): String {
    return if (input == null || input.text == null) "" else getStringSafe(input.text.toString())
}

fun getStringSafe(input: String): String {
    return if (isNullOrEmpty(input)) {
        ""
    } else input
}

fun getStringSafe(input: String, replacement: String): String {
    return if (isNullOrEmpty(input)) {
        replacement
    } else input
}

fun isValidEmail(target: EditText?): Boolean {
    return target != null && android.util.Patterns.EMAIL_ADDRESS.matcher(target.text).matches()
}

fun isValidEmail(target: CharSequence?): Boolean {
    return target != null && android.util.Patterns.EMAIL_ADDRESS.matcher(target).matches()
}

fun isValidPhoneNumber(phoneNumber: CharSequence): Boolean {
    return !TextUtils.isEmpty(phoneNumber) && android.util.Patterns.PHONE.matcher(phoneNumber).matches()
}

fun isValidWebsite(target: EditText): Boolean {
    return !isNullOrEmpty(target) && Patterns.WEB_URL.matcher(target.text).matches()
}

fun isValidWebsite(target: CharSequence): Boolean {
    return !TextUtils.isEmpty(target) && Patterns.WEB_URL.matcher(target).matches()
}

fun passwordCharValidation(password: String): Boolean {
    if (password.length < 8) {
        return false
    }
    val PASSWORD_PATTERN = "^(?=.*[A-Z])(?=.*[@_.]).*$"
    val pattern = Pattern.compile(PASSWORD_PATTERN)
    val matcher = pattern.matcher(password)
    return !password.matches(".*\\d.*".toRegex()) || !matcher.matches()
}

fun passwordValidationShort(password: String): Boolean {
    return password.length > 8
}
